packages/cli/src/cmds/validator/keys.ts

```diff
@@ -59,9 +59,19 @@ export async function getLocalSecretKeys(
     }
 
     const secretKeys = await Promise.all(
-      keystorePaths.map(async (keystorePath) =>
-        SecretKey.fromBytes(await Keystore.parse(fs.readFileSync(keystorePath, "utf8")).decrypt(passphrase))
-      )
+      keystorePaths.map(async (keystorePath) => {
+        const keystoreStr = fs.readFileSync(keystorePath, "utf8");
+
+        let keystore;
+        try {
+          keystore = Keystore.parse(keystoreStr);
+        } catch (e) {
+          (e as Error).message = `Error parsing keystore at ${keystorePath}: ${(e as Error).message}`;
+          throw e;
+        }
```

Comment:
@wemeetagain changed to only wrap the .parse step, not the file read. Also extended error to preserve stack trace